create_lmodsitepackage.py

```diff
@@ -210,6 +210,73 @@
     end
 end
 
+local function using_eessi_accel_stack ()
+    local modulepath = os.getenv("MODULEPATH") or ""
+    local accel_stack_in_modulepath = false
+
+    -- Check if we are using an EESSI version 2023 accelerator stack by checking if the $MODULEPATH contains
+    -- a path that starts with /cvmfs/software.eessi.io and contains accel/nvidia/ccNN
+    for path in string.gmatch(modulepath, '(.-):') do
+        if string.sub(path, 1, 41) == "/cvmfs/software.eessi.io/versions/2023.06" then
+            if string.find(path, "accel/nvidia/cc%d%d") then
+                accel_stack_in_modulepath = true
+                break
+            end
+        end
+    end
+    return accel_stack_in_modulepath
+end
+
+local function eessi_removed_module_warning_startup_hook(usrCmd)
+    if usrCmd == 'load' then
```

Comment:
Perhaps we should allow people to override this check completely if it turns out to be problematic? Via an environment variable?

Reply:
Good point, added that in https://github.com/EESSI/software-layer/pull/962/commits/9b4bed69d4258654407d9958f323ceb19d5eab28.

Reply:
```
$ ml ipympl/0.9.3-foss-2023a CUDA/12.1.1
Lmod has detected the following error: 
CUDA/12.1.1: All CUDA installations and modules depending on CUDA have been relocated to GPU-specific stacks.
Please see https://www.eessi.io/docs/site_specific_config/gpu/ for more information.

ipympl/0.9.3-foss-2023a: This module has been replaced by ipympl/0.9.3-gfbf-2023a

If you know what you are doing and you want to ignore this check for removed/relocated modules, set $EESSI_SKIP_REMOVED_MODULES_CHECK to any value. 


$ export EESSI_SKIP_REMOVED_MODULES_CHECK=1
$ ml ipympl/0.9.3-foss-2023a CUDA/12.1.1
Lmod has detected the following error:  The following module(s) are unknown: "ipympl/0.9.3-foss-2023a"
```